invenio_records_resources/services/records/service.py

```diff
@@ -93,8 +93,7 @@ def create_search(self, identity, record_cls, action='read',
         )
 
         # Avoid query bounce problem
-        if preference:
-            search = search.with_preference_param()
+        search = search.with_preference_param(preference)
 
         # Add document version to ES response
         search = search.params(version=True)
```

Comment:
Since we have changed operations on invenio-search we can simply chain them now:
```suggestion
        search = search.with_preference_param(preference).params(version=True)
```

Reply:
sure